xbmc/addons/AddonSystemSettings.cpp

```diff
@@ -121,4 +121,11 @@ int CAddonSystemSettings::GetAddonAutoUpdateMode() const
   return CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
       CSettings::SETTING_ADDONS_AUTOUPDATES);
 }
+
+AddonTrustedUpdateMode CAddonSystemSettings::GetAddonPreferredUpdateMode() const
+{
+  const int updateMode = CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
+      CSettings::SETTING_ADDONS_UPDATEMODE);
+  return static_cast<AddonTrustedUpdateMode>(updateMode);
+}
```

Comment:
I'm beginning to think the correct way to convert this is via a switch statement or an if statement.

```cpp
    switch (updateMode)
    {
        case static_cast<int>(AddonTrustedUpdateMode::PREFER_TRUSTED):
            return AddonTrustedUpdateMode::PREFER_TRUSTED;
        case static_cast<int>(AddonTrustedUpdateMode::ANY_REPOSITORY):
            return AddonTrustedUpdateMode::ANY_REPOSITORY;
        default :
            return AddonTrustedUpdateMode::PREFER_TRUSTED;
    }
```

or

```cpp
if (updateMode == static_cast<int>(AddonTrustedUpdateMode::ANY_REPOSITORY)
  return AddonTrustedUpdateMode::ANY_REPOSITORY;

// default
return AddonTrustedUpdateMode::PREFER_TRUSTED;
```

Reply:
yes, `if` looks cleaner i think. and there might/will come more options over time.

Reply:
i just wonder if we should really do that in this case. because usually a missing int cannot return. but i‘ll leave that decision to you. my personal feeling is it‘s a bit overkill 🙂

Reply:
It’s possible that the setting manager only allows the values from the settings.xml anyway. What you have is probably fine unless anyone else chimes in to say otherwise.

Reply:
how about:
```
const int updateMode = CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
      CSettings::SETTING_ADDONS_UPDATEMODE);
if (updateMode == undefined .... will figure what that is)
  AddonTrustedUpdateMode::PREFER_TRUSTED;
else
  return static_cast<AddonTrustedUpdateMode>(updateMode);
```

that wouldn't have to go under maintenance if an additional option comes in?

Reply:
That’s the problem with C++ enums. There is no way to check undefined easily.

I think we should assume that settings will only return valid options and leave it as the static cast unless someone objects!

Reply:
ok